src/controllers/discussion-controller.js

```diff
@@ -0,0 +1,168 @@
+        }
+        else{
+            responseNotFound(response);
+        }
+    })
+    .catch(function (error){
+        responseServer(response, error);
+    });
+}
+
+const getDiscussionsCriterion = async (request, response) => {
+    const filter = request.params.filter;
+    const criterion = request.params.criterion;
+    if(filter == "title"){
+        Discussions.find({title:{$regex : new RegExp(criterion, "i")}}, {title:1, dateCreation:1, numberComments:1, theme:1})
+        .then(function (discussions) {  
+            if(discussions.length){
+                response.status(StatusCodes.OK).json(discussions);
+            }
+            else{
+                responseNotFound(response);
+            }
+        })
+        .catch(function (error){
+            responseServer(response, error);
+        });
+    } 
+    else{
+        Accounts.findOne({_id:criterion}, {_id:0, discussions:1})
+        .then(async (account) =>{
+            if(!account || account.discussions == '[]'){
+                responseNotFound(response);
+            }
+            else{
+                const disccussions = await Discussions.find({_id:{ $in: account.discussions}}, {title:1, dateCreation:1, numberComments:1, theme:1})
+                if(disccussions.length){
+                    response.status(StatusCodes.OK).json(disccussions);
+                }
+                else{
+                    responseNotFound(response);
+                }
+            }
+        })
+        .catch(function (error){
+            responseServer(response, error);
+        }); 
+    } 
+}
+
+const getDiscussions = async (request, response) => {
+    Discussions.find(null,{title:1, dateCreation:1, numberComments:1, theme:1})
+    .then(function (discussions) {  
+        if(discussions.length){
+            response.status(StatusCodes.OK).json(discussions);
+        }
+        else{
+            responseNotFound(response);
+        }
+    })
+    .catch(function (error){
+        responseServer(response, error);
+    });
+}
+
+const getDiscussion = async (request, response) => {
+    const discussionID = request.params.discussionID;
+    Discussions.findById(discussionID)
+    .populate({path: 'idAccount', select: 'name lastname'})
+    .then(function (discussion) {  
+        if(discussion){
+            response.status(StatusCodes.OK).json(discussion);
+        }
+        else{
+            responseNotFound(response);
+        }
+    })
+    .catch(function (error){
+        responseServer(response, error);
+    });
+}
+
+
+const postDiscussion = async (request, response) => {
+    const {
+        title,comment, theme, idAccount
+    } = request.body;
+    const idAccountConverted  = mongoose.Types.ObjectId(idAccount);
+    const dateNow = new Date();
+    const dateCreation = new Date(dateNow.getTime() - (dateNow.getTimezoneOffset() * 60000 )).toISOString().slice(0, 10);
+    const discussion = new Discussions ({
+        title: title,
```

Comment:
Codacy has a fix for the issue: [Expected property shorthand.](https://app.codacy.com/gh/Martha-Miroslava/APIBiziidiDiidxaza/pullRequest?prid=8267260)
```suggestion
        title,
```